### PR TITLE
add `checkouts` connection to `User` type

### PR DESCRIPTION
**Pull request by zedzior**
https://github.com/saleor/saleor/issues/10818

I want to merge this change in order to add `checkouts` connection to `me` query with ability to sort checkouts by channel.

**The query looks like:**
```
query {
    me {
        checkouts (first: 1, channel: "default-channel") {
            edges {
                node {
                    id
                }
            }
        }
    }
}
```

**Expected result:**
```
"data": {
    "me": {
        "checkouts": {
            "edges": [
                {
                    "node": {
                        "id": "Q2hlY2tvdXQ6YjJjZDc3MDQtMGY2NS00OWQ5LWExMGItMzU3YTQ5NDA5OWZi"
                    }
                }
            ]
        }
    }
},
```
 

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by kadewu — COMMENTED** [review]
Please update graphql schema. You can run locally `npm run build-schema` to do so. You should as well start using `pre-commit`.

File: saleor/graphql/account/types.py [hunk 1]
```diff
@@ -271,6 +271,13 @@ class User(ModelObjectType):
             description="Slug of a channel for which the data should be returned."
         ),
     )
+    checkouts = ConnectionField(
+        CheckoutCountableConnection,
+        description="Returns checkouts assigned to this user.",
```
Comment (maarcingebala) [inline saleor/graphql/account/types.py:276]: We need to mention version in which it was added, see usages of ADDED_IN_38 and here as well.

File: saleor/graphql/account/tests/test_account.py [hunk 2]
```diff
@@ -1125,6 +1145,43 @@ def test_me_query_checkout_with_inactive_channel(user_api_client, checkout):
...
+    response = user_api_client.post_graphql(
+        query, {"channel": checkout.channel.slug}
+    )
+
+    content = get_graphql_content(response)
+    data = content["data"]["me"]
+    assert data["checkouts"]["edges"][0]["node"]["id"] == graphene.Node.to_global_id(
+        "Checkout", checkout.pk
+    )
+    assert data["checkouts"]["totalCount"] == 1
```
Comment (maarcingebala) [inline saleor/graphql/account/tests/test_account.py:1183]: Would be also good to also check that the query returned the checkout from the right channel as well.

File: saleor/graphql/checkout/dataloaders.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def batch_load(self, keys):
         checkout_by_user_map = defaultdict(list)
         for checkout in checkouts:
             checkout_by_user_map[checkout.user_id].append(checkout)
-        return [checkout_by_user_map.get(user_id) for user_id in keys]
+        return [checkout_by_user_map.get(user_id, []) for user_id in keys]
```
Comment (fowczarek) [inline saleor/graphql/checkout/dataloaders.py:173]: It's `defaultdict(list)` that means `[]` is default value if key do not exists. 
```suggestion
        return [checkout_by_user_map.get(user_id) for user_id in keys]
```

File: saleor/graphql/checkout/dataloaders.py [hunk 4]
```diff
@@ -192,7 +192,7 @@ def batch_load(self, keys):
         for checkout in checkouts:
             key = (checkout.user_id, checkout.channel_slug)
             checkout_by_user_and_channel_map[key].append(checkout)
-        return [checkout_by_user_and_channel_map.get(key) for key in keys]
+        return [checkout_by_user_and_channel_map.get(key, []) for key in keys]
```
Comment (fowczarek) [inline saleor/graphql/checkout/dataloaders.py:195]: It's `defaultdict(list)` that means `[]` is default value if key do not exists. 
```suggestion
        return [checkout_by_user_and_channel_map.get(key) for key in keys]
```

**Review by fowczarek — COMMENTED** [review]
Should you add a test for checkouts pagination on `me` query?

File: saleor/graphql/checkout/dataloaders.py (hunk 3 above)
Comment (zedzior) [reply saleor/graphql/checkout/dataloaders.py:173]: Actually, when there is no single checkout, the `[checkout_by_user_map.get(user_id) for user_id in keys]` expression will evaluate to `[None]`. With current implementation it returns `[[]]`. 
Later on, in `create_connection_slice` function, we check `if isinstance(iterable, list)`, so I kept current implementation.

Comment (fowczarek) [reply saleor/graphql/checkout/dataloaders.py:173]: Maybe we shouldn't use get on `defatultdict`?
```suggestion
        return [checkout_by_user_map[user_id] for user_id in keys]
```
```python
In [1]: from collections import defaultdict

In [2]: checkout_by_user_map = defaultdict(list)

In [3]: checkout_by_user_map.get("fake_key")

In [4]: checkout_by_user_map["fake_key"]
Out[4]: []
```

Comment (zedzior) [reply saleor/graphql/checkout/dataloaders.py:173]: nice! thanks

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by kadewu — APPROVED** [review]

**Review by kadewu — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**